CRM/Contact/BAO/GroupContact.php

```diff
@@ -211,6 +213,12 @@ public static function removeContactsFromGroup(
           'tracking' => $tracking,
         ];
         CRM_Contact_BAO_SubscriptionHistory::create($historyParams);
+        // Removing a row from civicrm_group_contact for a smart group may mean a contact
+        // Is now back in a group based on criteria so we should re-load the cache.
```

Comment:
Actually it's never a good idea to pro-actively load a cache in my experience - it will get loaded if needed